remove errant pragmas inside functions that older gcc versions do not support

diff --git example/x11.c example/x11.c
index c49b64e..01b0c7d 100644
--- example/x11.c
+++ example/x11.c
@@ -325,12 +325,9 @@ main (int argc, char *argv[])
         libssh2_trace(session, LIBSSH2_TRACE_CONN);
 
     /* ignore pedantic warnings by gcc on the callback argument */
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wpedantic"
     /* Set X11 Callback */
     libssh2_session_callback_set(session, LIBSSH2_CALLBACK_X11,
                                  (void *)x11_callback);
-#pragma GCC diagnostic pop
 
     /* Authenticate via password */
     rc = libssh2_userauth_password(session, username, password);
